Handle dashes in repo name and add underscore replacement logic

LingEnOwO opened the pull request:
### Summary
This PR addresses an issue where using dashes in the import path causes generated psql commands in Taskfile.yml to fail. It:

- Validates that `repoName` only includes alphanumerics, dashes, or underscores
- Adds a `repoNameUnderscores` value for replacing `kb_example` placeholders
- Updates the template (kbexample/Taskfile.yml) to use `kb_example` in DB commands

michaelshuh commented on gen.go:
```diff
@@ -44,6 +45,11 @@ var (
+			if !regexp.MustCompile(`^[A-Za-z0-9_-]+$`).MatchString(repoName) {
+				return fmt.Errorf("repository name can only contain letters, numbers, dashes or underscores")
+			}
+			repoNameUnderscores := strings.ReplaceAll(repoName, "-", "_")
```
This works - but this could also be tested, not sure if there are maybe other things in the future where you may want to have some method like `CleanRepoName` and then test it

michaelshuh reviewed (commented):
Overall looks good! Just a few comments

LingEnOwO commented:
Thanks for the feedback! I’ve extracted the regex into `ValidateRepoName`, switched the tests to use `testify/assert`, and updated `go.mod` and `go.sum`
